services/sidecar/src/simcore_service_sidecar/celery_configurator.py

```diff
@@ -7,217 +7,100 @@
 use a look ahead function to check the type of upcoming task and
 schedule it accordingly.
 """
-from pprint import pformat
-from typing import Optional, Tuple
+from functools import wraps
+from typing import Callable
 
-from celery import Celery, states
-from simcore_sdk.config.rabbit import Config as RabbitConfig
+from celery import Celery
+from celery.contrib.abortable import AbortableTask
+from kombu import Queue
 
 from . import config
 from .boot_mode import BootMode, get_boot_mode, set_boot_mode
 from .celery_log_setup import get_task_logger
-from .cli import run_sidecar
-from .core import task_required_resources
-from .utils import assemble_celery_app, is_gpu_node, start_as_mpi_node, wrap_async_call
+from .celery_task import entrypoint
+from .celery_task_utils import on_task_failure_handler, on_task_success_handler
+from .utils import is_gpu_node, start_as_mpi_node
 
 log = get_task_logger(__name__)
 
 
-# used by internal queues in this module
-_rabbit_config = RabbitConfig()
-_celery_app_cpu = assemble_celery_app("celery", _rabbit_config)
-_celery_app_gpu = assemble_celery_app("celery_gpu_mode", _rabbit_config)
-_celery_app_mpi = assemble_celery_app("celery_mpi_mode", _rabbit_config)
+CELERY_APP_CONFIGS = {
+    BootMode.CPU: {"app": "celery_cpu_mode", "queue_name": config.CPU_QUEUE_NAME},
+    BootMode.GPU: {"app": "celery_gpu_mode", "queue_name": config.GPU_QUEUE_NAME},
+    BootMode.MPI: {"app": "celery_mpi_mode", "queue_name": config.MPI_QUEUE_NAME},
+}
 
 
-def dispatch_comp_task(user_id: str, project_id: str, node_id: str) -> None:
-    """Uses the director's API to determineate where the service needs
-    to be dispacted and sends it to the appropriate queue"""
-    # TODO: use _node_id to check if this service needs a GPU or NOT, ask director
-    # then schedule to the correct queue
-    # Add logging at #TODO: #1 and here to make sure the task with the same uuid is scheduled on the correct worker!
-    if node_id is None:
-        log.error("No node_id provided for project_id %s, skipping", project_id)
-        return
+def celery_adapter(app: Celery) -> Callable:
```

Comment:
FYI: I think this adapter thingy (also used in computation_subscribe) can be really simplified by doing a make function that binds the argument, i.e. in ``create_tasks.py`` rewrite ``entrypoint`` as  
```python

def create_task_entrypoint(app: Celery) -> Coroutine: 
      def entrypoint(zelf, *, user_id: str, project_id: str, node_id: Optional[str] = None) -> None:
              log.info("Received task %s", self.request.id)
              _shared_task_dispatch(zelf, app, user_id, project_id, node_id)
              log.info("Completed task %s", zelf.request.id)
      return entrypoint
```